### PR TITLE
Fix batch norm folding in `prepare_pt2e` for multiple conv->BN chains sharing the same conv weights

### PR DESCRIPTION
**Pull request by subhankarpal**
### Summary
For a model with multiple conv->BN chains that have the same conv weights, `prepare_pt2e` applies batch norm folding repeatedly for each chain, causing incorrect conv weights in the output prepared model. This fix ensures batch norm folding is performed only once per unique conv weight set, preventing multiple folds and preserving correct model behavior.

### Testplan
```
python test/quantization/pt2e/test_quantize_pt2e.py -k test_chunked_bn_fusion
```

### CODE REVIEW

File: torchao/testing/model_architectures.py [hunk 1]
```diff
@@ -22,6 +22,26 @@ def forward(self, x):
         return x
 
 
+class ToyCNNModel(nn.Module):
```
Comment (jerryzh168) [inline torchao/testing/model_architectures.py]: nit: better to be more specific I think, like `ConvWithSharedWeightInExportedModel`

File: torchao/quantization/pt2e/utils.py [hunk 2]
```diff
@@ -781,9 +795,15 @@ def _fuse_conv_bn_(m: GraphModule) -> None:
...
+            conv_node,
+            conv_weight_node,
+            conv_bias_node,
+            bn_node,
+            m,
+            (conv_weight_node in fused_convs_weight_nodes)
         )
-
+        fused_convs_weight_nodes.add(conv_weight_node)
+    del fused_convs_weight_nodes
```
Comment (jerryzh168) [inline torchao/quantization/pt2e/utils.py]: nit: probably no need to delete this explicitly

**Comment by subhankarpal** [comment]
> see some nit comments inline

Fixed, thanks.

**Comment by jerryzh168** [comment]
@subhankarpal no need for manual import, our oncall will import this in diff train

**Comment by amdfaa** [comment]
This PR appears to be causing failures for both rocm and cuda: https://hud.pytorch.org/failure?name=Run%20Regression%20Tests%20on%20ROCm%20%2F%20test-nightly%20(ROCM%20Nightly%2C%20linux.rocm.gpu.gfx942.2%2C%20--pre%20torch%20--index-url%20https%3A%2F%2Fdownload.pyt...%20%2F%20linux-job&jobName=undefined&failureCaptures=test%2Fintegration%2Ftest_integration.py%3A%3ATestAutoQuant%3A%3Atest_autoquant_hp_float
